### PR TITLE
feat: Inject-ArmContent.ps1: support absolute paths to external files to inject

### PR DESCRIPTION
**Pull request by vromantp**
Added the possibility to have an absolute path pointing to an external file to be injected. Should obviously be backwards compatible with relative paths (from the 'parent' file)

Changes:
- updated PS script (both code and comment)
- Extended integration test (for both windows and linux). Note: only windows could be tested locally
- Updated docs

### CODE REVIEW

**Comment by stijnmoreels** [comment]
Great! 👍  Thx for this! I already created an issue to investigate a possible different input test generation file here: #279 

Pim and me will review it in the near future 😉 .

**Comment by pim-simons** [comment]
Awesome, many thanks for the contribution!! 👍🏻 
Will try to review this week or in week 13 because I am on holiday next week 🏂🏻

**Comment by stijnmoreels** [comment]
Hmm, I think the build fails here bc your forked master branch doesn't have yet the latest fixes on generating Azure Integration account private certificates (https://github.com/arcus-azure/arcus.scripting/pull/277). Can you merge with the upstream branch? Thx.

File: src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1
Comment (stijnmoreels) [inline src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1:13]:
```suggestion
param (

```

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.ARM.tests.ps1
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.ARM.tests.ps1:43]:
```suggestion
                    $armTemplate = Get-Content -path $armTemplateFile -Raw
                    $armTemplate = $armTemplate -replace '#{ArmTemplateDirectory}#', $armTemplateDirectory
                    $armTemplate | Set-Content -Path $armTemplateFile

```
Let's split this for better readability. Usually, we don't inline too much.

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.ARM.tests.ps1
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.ARM.tests.ps1:78]:
```suggestion
                    $armTemplate = Get-Content -path $armTemplateFile -Raw
                    $armTemplate = $armTemplate -replace '#{ArmTemplateDirectory}#', $armTemplateDirectory
                    $armTemplate | Set-Content -Path $armTemplateFile

```
Let's split this for better readability. Usually, we don't inline too much.

File: src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1
Comment (stijnmoreels) [inline src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1:122]: Hihi, thanks for this 😉 .

**Review by stijnmoreels — APPROVED** [review]
Thx very much for this, Pieter! 💯 Is a great addition to the existing functionality. Great job in adding tests for this and providing us with the necessary test files.

A tip I want to give you in GitHub forking, use a separate dev branch on your fork instead of your `master` branch. It will help a lot when keeping your fork up to date instead of constantly worrying about merge conflicts. See [the section about forking in our contribution guide](https://github.com/arcus-azure/.github/blob/master/CONTRIBUTING.md#forking).

Saw that the test files are having the right line ending characters, so that's very good! 👍

File: docs/preview/02-Features/powershell/arm.md
Comment (stijnmoreels) [inline docs/preview/02-Features/powershell/arm.md:67]: I'm wondering if '... path relative to the parent file...' is clear enough. Maybe @pim-simons can shed some light here.

Comment (pim-simons) [reply docs/preview/02-Features/powershell/arm.md:67]: Combined with the examples on line 71 to 74 this is clear enough for me 👍🏻

**Review by pim-simons — CHANGES_REQUESTED** [review]
Great work! 💯  I don't really have anything to add that @stijnmoreels  hasn't already mentioned 😉 

As @stijnmoreels mentioned (https://github.com/arcus-azure/arcus.scripting/pull/280#issuecomment-1070369976) the build fails because a recent fix isn't yet in your branch, merging with upstream should fix that.

I do see other tests failing though with this error "RuntimeException: Method invocation failed because [System.IO.Path] does not contain a method named 'IsPathFullyQualified'"
- Arcus ARM integration tests. ARM injection. Replaces file path with inline file contents
- Arcus ARM integration tests. ARM injection. Replaces relative file path with file contents as JSON object (windows)
- Arcus ARM integration tests. ARM injection. Replaces absolute file path with file contents as JSON object (windows)
- Arcus ARM integration tests. ARM injection. Replaces file path with file contents as escaped JSON and replaced special characters

Once that error is fixed we can merge to main 👍🏻

File: docs/preview/02-Features/powershell/arm.md
Comment (stijnmoreels) [reply docs/preview/02-Features/powershell/arm.md:67]: Ok, great!

**Comment by vromantp** [comment]
As it turns out `[System.IO.Path]::IsPathFullyQualified()` only available on Windows in PS 7+, not in PS 5.1, so I had to re-implement somehow.

File: src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1
Comment (stijnmoreels) [inline src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1:25]: Why do we make a distinction here between blank <> empty? Is this documented somehow that this indeed describes different results in IO?

File: src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1
Comment (stijnmoreels) [inline src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1:19]: It's a good thing that we have this, then, if it's not supported. I'm wondering though, if this is still good to follow. We're toggling booleans twice just for the sake of this method to be called `IsPathFullyQualified`.

I'll let @pim-simons have a look. I was thinking if it were something in the like of `IsPathRelative` it wouldn't need to to be switched with `-not (-not ( ) )`.

**Review by stijnmoreels — CHANGES_REQUESTED** [review]
Some small things to check, but otherwise we're good to go! 🥇 
Thx for looking into this and detecting the problem here with PowerShell versions. Great stuff.

**Comment by vromantp** [comment]
Turns on that the Windows-implementation of `Path.IsPathFullyQualified` is basically [Pandora's Box](https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/IO/PathInternal.Windows.cs) compared to the [Linux-logic](https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/IO/PathInternal.Unix.cs). 

This means that my custom function is not remotely 'functionally equal', so I'll rework and get back to you with an alternative implementation :)

**Comment by pim-simons** [comment]
> Turns on that the Windows-implementation of `Path.IsPathFullyQualified` is basically [Pandora's Box](https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/IO/PathInternal.Windows.cs?rgh-link-date=2022-03-18T09%3A06%3A30Z) compared to the [Linux-logic](https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/IO/PathInternal.Unix.cs?rgh-link-date=2022-03-18T09%3A06%3A30Z).
> 
> This means that my custom function is not remotely 'functionally equal', so I'll rework and get back to you with an alternative implementation :)
> 
> For those interested: the specfics of `IsPathFullyQualified` can be found in the 'Remarks' section here: [docs.microsoft.com/en-us/dotnet/api/system.io.path.ispathfullyqualified?view=net-6.0](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.ispathfullyqualified?view=net-6.0)

Thanks for your work!! Greatly appreciated 👍🏻

**Comment by vromantp** [comment]
This new implementation is an improvement IMO. Integration tests are succeeding now as well.

File: src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1
Comment (stijnmoreels) [inline src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1:53]: Great! This is definitely an improvement on all fronts. Nice job! 👍

**Review by stijnmoreels — APPROVED** [review]
Great stuff! 🥇

**Pull request merged**
